sdk/src/binary.rs

```diff
@@ -196,6 +201,44 @@ impl VariableLengthBinary for Binary {
     }
 }
 
+impl Debug for Binary {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "Binary(")?;
+        let mut iter = self.iter();
+        if let Some(x) = iter.next() {
+            write!(f, "{:?}", x)?;
+        }
+        for x in iter {
+            write!(f, ", {:?}", x)?;
+        }
+        write!(f, ")")?;
+        Ok(())
+    }
+}
+
+#[derive(Clone)]
+pub struct BinIter(Binary);
+
+impl Iterator for BinIter {
+    type Item = u8;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.0.len() == 0 {
+            None
+        } else {
+            let val = self.0.env().binary_front(self.0 .0.to_object());
+            self.0 = self.0.slice(1..);
+            let val = unsafe { <u32 as RawValConvertible>::unchecked_from_val(val) } as u8;
+            Some(val)
+        }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let len = self.0.len() as usize;
+        (len, Some(len))
+    }
+}
```

Comment:
Could you also implement `DoubleEndedIterator`, `ExactSizeIterator`, and `FusedIterator`? See `VecIter` for an example:
https://github.com/stellar/rs-stellar-contract-sdk/blob/7c3f909d519393cad5d11b54ace704e13e619b12/sdk/src/vec.rs#L416-L444

Reply:
Done.